config/v1/types.go

```diff
@@ -398,3 +398,13 @@ const (
 	// NoOpinionIncludeSubDomains means HSTS "includeSubDomains" doesn't matter to the RequiredHSTSPolicy
 	NoOpinionIncludeSubDomains IncludeSubDomainsPolicy = "NoOpinion"
 )
+
+// GroupVersionResource includes the group, version and type of resource
+type GroupVersionResource struct {
```

Comment:
You should move this to the operator package, we don't allow cross API group references like this

Please also rename it `PinnedResourceReference`

I'd also like to see field validations each of the fields, required vs optional, examples of valid values, patterns for the values where appropriate.

Please review the [API conventions doc](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc) and review the section about good godocs to help improve the context here